nix/nix.go

```diff
@@ -131,22 +133,30 @@ func (ctx *NixContext) GetMachines(deploymentPath string) (hosts []Host, err err
 	return hosts, nil
 }
 
-func (ctx *NixContext) BuildMachines(deploymentPath string, hosts []Host, nixArgs []string, nixBuildTargets string) (path string, err error) {
+func (ctx *NixContext) BuildMachines(deploymentPath string, hosts []Host, nixArgs []string, nixBuildTargets string) (resultPath string, err error) {
 	hostsArg := "["
 	for _, host := range hosts {
 		hostsArg += "\"" + host.TargetHost + "\" "
 	}
 	hostsArg += "]"
 
-	// create tmp dir for result link
-	tmpdir, err := ioutil.TempDir("", "morph-")
-	if err != nil {
-		return "", err
+	resultLinkPath :=
+	  fmt.Sprintf("%s/.gcroots/%s",
```

Comment:
I'm not entirely sure this is what we want. I guess it's about tradeoffs. Either you'd place the result link relative to morph cwd, or you'd place it relative to the deployment path. You've done the latter. Nix-build does the former. 

I'd personally prefer that we just add flags to morph `--out-link=<path>` (default: cwd) and `--no-out-link` (default: false) and propagate those more or less directly to nix-build. This would yield the familiar and recognizable "./result"-link and would align morph better with the Nix CLI.

Reply:
Since cwd changes you can end up with old versions littered about the place (I hardly ever call it from the same directory), I wanted in the same place so it gets replaced (and doesn't start to resemble generations). Additionally using `result` is a problem if you use morph to build several nix-files in the same directory (e.g. k8s.nix, ceph.nix).

`--no-out-link` instead of a temp file requires parsing the nix-build output, which I guess is what the person who wrote this code didn't want to do. If we did that, then you could just pass out-link through morph and leave it to the user to think about GCing, though I feel in this instance it's nice of morph to just more or less do the right thing.

Reply:
Makes sense. And I see why you made that choice then. 
I still have some basic demands/suggestions though:

1. Build shouldn't abort if gc-root creation fails. (perhaps just a warning on stderr?)

2. It should be possible to disable this.
 2.1. Should it actually be default enabled or disabled? I thiiiink... I'm fine with enabled, but that _definitely_ requires `1.`. @adamtulinius can you see any issues here?

3. To make the path more configurable, perhaps we could convert the bool flag to a path flag e.g `--gc-root-path=./.gcroots` (feel free to name-bikeshed); where the value is relative to the deployment path and which, if empty, disables link creation?

Reply:
>     2.1. Should it actually be default enabled or disabled?

I would prefer disabled by default, because of the principle of least surprise. I know nix-build makes links by default, but that's.. Annoying, imo. If people want this behaviour by default, they can wrap morph (like we do ourselfs).

>     3. To make the path more configurable, perhaps we could convert the bool flag to a path flag e.g `--gc-root-path=./.gcroots`

I like this, but would prefer if the path could be both absolute and relative. Also, I think we should strip the `.nix`-extension from the name of the gcroot.

Reply:
I don't think POLA applies here, but it deviates from previous behavior so defaulting to off is sensible. A path relative to the nix-file instead of cwd might be surprising though.

Reply:
Generally I have much fewer demands to a feature, if it doesn't change the default behavior. :-) I can live with how it is now, and then let's see what feedback we get.

Reply:
> I don't think POLA applies here

You could just have said "makes sense", you know. :P

Reply:
> A path relative to the nix-file instead of cwd might be surprising though.

I should have been more clear. I meant relative as in the current solution. So relative to the deployment files.

Reply:
Btw. please use path.Join instead of fmt.Sprintf for creating paths :-)

Reply:
d1ed927